tests/functional/adapter/test_python_submissions.py

```diff
@@ -1,89 +1,177 @@
-
-#     @pytest.fixture(scope="class")
-#     def models(self):
-#         return {"model.sql": basic_sql, "incremental.py": incremental_python}
-
-#     def test_incremental(self, project):
-#         vars_dict = {
-#             "test_run_schema": project.test_schema,
-#         }
-
-#         results = run_dbt(["run", "--vars", yaml.safe_dump(vars_dict)])
-#         assert len(results) == 2
+import os
+import shutil
+from collections import Counter
+from copy import deepcopy
+
+import pytest
+import yaml
+
+from dbt.tests.adapter.python_model.test_python_model import (
+    BasePythonIncrementalTests,
+    BasePythonModelTests,
+)
+from dbt.tests.util import run_dbt
+
+basic_sql = """
+{{ config(materialized="table") }}
+select 1 as column_1, 2 as column_2, '{{ run_started_at.strftime("%Y-%m-%d") }}' as run_date
+"""
+
+basic_python = """
+def model(dbt, spark):
+    dbt.config(
+        materialized='table',
+    )
+    df =  dbt.ref("model")
+    return df
+"""
+
+basic_spark_python = """
+def model(dbt, spark_session):
+    dbt.config(materialized="table")
+
+    data = [(1,), (2,), (3,), (4,)]
+
+    df = spark_session.createDataFrame(data, ["A"])
+
+    return df
+"""
+
+second_sql = """
+select * from {{ref('my_python_model')}}
+"""
+
+schema_yml = """version: 2
+models:
+  - name: model
+    versions:
+      - v: 1
+"""
+
+
+class TestBasePythonModelTests(BasePythonModelTests):
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "schema.yml": schema_yml,
+            "model.sql": basic_sql,
+            "my_python_model.py": basic_python,
+            "spark_model.py": basic_spark_python,
+            "second_sql_model.sql": second_sql,
+        }
+
+
+incremental_python = """
+def model(dbt, spark_session):
+    dbt.config(materialized="incremental")
+    df = dbt.ref("model")
+
+    if dbt.is_incremental:
+        max_from_this = (
+            f"select max(run_date) from {dbt.this.schema}.{dbt.this.identifier}"
+        )
+        df = df.filter(df.run_date >= spark_session.sql(max_from_this).collect()[0][0])
+
+    return df
+"""
+
+
+class TestBasePythonIncrementalTests(BasePythonIncrementalTests):
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"models": {"+incremental_strategy": "append"}}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {"model.sql": basic_sql, "incremental.py": incremental_python}
+
+    def test_incremental(self, project):
+        vars_dict = {
+            "test_run_schema": project.test_schema,
+        }
+
+        results = run_dbt(["run", "--vars", yaml.safe_dump(vars_dict)])
+        assert len(results) == 2
+
+
+class TestPythonClonePossible:
+    """Test that basic clone operations are possible on Python models. This
+    class has been adapted from the BaseClone and BaseClonePossible classes in
+    dbt-core and modified to test Python models in addition to SQL models."""
```

Comment:
I thought about inheriting from `BaseClone` ([source here](https://github.com/dbt-labs/dbt-core/blob/d21ff38ba1aafdee67e0165856f59b31d824d144/tests/functional/adapter/dbt_clone/test_dbt_clone.py#L25-L98)) instead of reimplementing its fixtures and helper functions, but there are enough differences in implementation required to support Python models that I decided the maintenance risk incurred by extending a class that could well change underneath us in the future outweighed the benefit of fewer lines of code. Happy to revisit this decision if you have a different preference, or to use a different base class, but as far as I could tell `dbt-core` doesn't yet have a base class to test this particular functionality (i.e. cloning a Python model).